2_correlation_analysis/spell_vs_counts_experiment/1_correlation_analysis.py

```diff
@@ -0,0 +1,253 @@
+import matplotlib.pyplot as plt
+import umap
+import random
+import numpy as np
+import scipy
+from scipy.spatial.distance import pdist, squareform
+from sklearn.decomposition import PCA
+from scripts import paths, utils
+
+# ## Set user parameters
+#
+# Here we set the number of PCs or singular vectors to use. We are starting with 300 since this is what [eADAGE](https://pubmed.ncbi.nlm.nih.gov/28711280/) used.
+
+# +
+# Params
+
+# Which subset of genes to consider: core, acc, all
+subset_genes = "all"
+
+# The number of accessory genes is 200 - 500
+# The number of core genes is ~ 5000
+# So the number of singular vectors is relative to the number of genes
+if subset_genes == "acc":
+    num_SVs = 50
+else:
+    num_SVs = 100
+# -
+
+# Load expression data
+pao1_compendium_filename = paths.PAO1_COMPENDIUM
+pa14_compendium_filename = paths.PA14_COMPENDIUM
+
+pao1_compendium = pd.read_csv(pao1_compendium_filename, sep="\t", header=0, index_col=0)
+pa14_compendium = pd.read_csv(pa14_compendium_filename, sep="\t", header=0, index_col=0)
+
+print(pao1_compendium.shape)
+pao1_compendium.head()
+
+print(pa14_compendium.shape)
+pa14_compendium.head()
+
+# ## Get core/accessory genes
+#
+# We will subset the correlation matrix to only consider core genes.
+#
+# _Rationale:_ Previously we used all genes (both core and accessory) to create a co-expression network, but due to the large imbalance in the number of core genes compared to accessory genes, no module was found to be "mostly core." Instead we will perform separate analyses of core and accessory genes to examine co-expression patterns.
+
+# +
+# Read in expression data
+pao1_expression_filename = paths.PAO1_COMPENDIUM
+pa14_expression_filename = paths.PA14_COMPENDIUM
+
+pao1_expression = pd.read_csv(pao1_expression_filename, sep="\t", index_col=0, header=0)
+pa14_expression = pd.read_csv(pa14_expression_filename, sep="\t", index_col=0, header=0)
+
+# +
+pao1_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_annot_filename = paths.GENE_PA14_ANNOT
+
+core_acc_dict = utils.get_my_core_acc_genes(
+    pao1_annot_filename, pa14_annot_filename, pao1_expression, pa14_expression
+)
+# -
+
+pao1_core = core_acc_dict["core_pao1"]
+pa14_core = core_acc_dict["core_pa14"]
+pao1_acc = core_acc_dict["acc_pao1"]
+pa14_acc = core_acc_dict["acc_pa14"]
+
+# ## Select gene subset
+
+# Select subset of genes
+if subset_genes == "core":
+    pao1_compendium = pao1_compendium[pao1_core]
+    pa14_compendium = pa14_compendium[pa14_core]
+elif subset_genes == "acc":
+    pao1_compendium = pao1_compendium[pao1_acc]
+    pa14_compendium = pa14_compendium[pa14_acc]
+
+print(pao1_compendium.shape)
+print(pa14_compendium.shape)
+
+# ## Correlation of raw gene expression data
+#
+# Here is the correlation of the raw data without any malnipulations. This will serve as a reference to compare the correlations below where applied corrections to the correlations to account for the dominant signal described above.
```

Comment:
```suggestion
# Here is the correlation of the raw data without any manipulations. This will serve as a reference to compare the correlations below where applied corrections to the correlations to account for the dominant signal described above.
```